### ISSUE TITLE
Make parantheses optional.

### ISSUE THREAD

**Issue by AnthonyJacob** [lexi-lambda/rascal#13]
I have not much experience with Lispy languages, but there was always something that bugged me about its syntax: Why do I need to write parentheses when they can be inferred? 

Example : 

``` lisp
(def foo (a b)
	(doSomething a b)
)
```

The **indentation** makes it obvious that `doSomething a b` is member of `(def foo ...`. Lets take a look how what happens when we hide all `()` that can be inferred at compile time.

``` lisp
def foo (a b)
	doSomething a b
```

*That is much more pleasant to look at.*

In other words, why not make the syntax indentation sensitive? Can it be done in racket?

There is an interesting project [Oczor](https://github.com/ptol/oczor) that does exactly that. Except it uses records instead of lists by default.

**Comment by lexi-lambda** [comment]
Since the reader can be customized, this could be implemented as a `#lang` that internally reads or expands to `#lang rascal`. See [`sweet-exp`](http://docs.racket-lang.org/sweet/index.html) for an example.

However, generally doing this is very difficult to do in general in a pleasant way for non-trivial examples when macros are at play, and I personally don’t find the syntax difference a large improvement. With a good editor and a little experience reading Lisp code, the parentheses just “fade away”. For now, I’m going to close this as “won’t fix”, but it might be worth revisiting at some point if this project gets more off the ground.
